src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -92,190 +48,13 @@
 public interface GameRegistry {
 
     /**
-     * Attempts to retrieve the specific type of {@link CatalogType} based on
-     * the string id given.
+     * Gets an instance of the provided {@link Provider} class.
      *
-     * <p>Some types may not be available for various reasons including but not
-     * restricted to: mods adding custom types, plugins providing custom types,
-     * game version changes.</p>
-     *
-     * @param typeClass The class of the type of {@link CatalogType}
-     * @param id The string id of the catalog type
-     * @param <T> The type of catalog type
-     * @return The found catalog type, if available
-     */
-    <T extends CatalogType> Optional<T> getType(Class<T> typeClass, String id);
```

Comment:
Same again for getType as `getProvider(CatalogTypeProvider.class).getType` is less obvious